pymatgen/entries/computed_entries.py

```diff
@@ -11,26 +11,187 @@
+
+
+class CompositionEnergyAdjustment(EnergyAdjustment):
+    """
+    An energy adjustment applied to a ComputedEntry based on the atomic composition.
+    Used in various DFT energy correction schemes.
+    """
+    def __init__(self, adj_per_atom, n_atoms, name, cls="None", description="Composition-based energy adjustment"):
+        """
+        Args:
+            adj_per_atom: float, energy adjustment to apply per atom, in eV/atom
+            n_atoms: float or int, number of atoms
+            name: str, human-readable name of the energy adjustment.
+                (Default: "")
+            cls: str, the name of the Compatibility class used to generate the
+                energy adjustment. (Default: None)
+            description: str, human-readable explanation of the energy adjustment.
+        """
+        self._value = adj_per_atom
+        self.n_atoms = n_atoms
+        self.cls = cls
+        self.name = name
+        self.description = description + " ({:.3f} eV/atom x {} atoms)".format(self._value,
+                                                                               self.n_atoms
+                                                                               )
+
+    @property
+    def value(self):
+        """
+        Return the value of the energy adjustment in eV.
+        """
+        return self._value * self.n_atoms
+
+
+class TempEnergyAdjustment(EnergyAdjustment):
+    """
+    An energy adjustment applied to a ComputedEntry based on the temperature.
+    Used, for example, to add entropy to DFT energies.
+    """
+    def __init__(self, adj_per_deg, temp, n_atoms, name="", cls="None", 
+                 description="Temperature-based energy adjustment"):
+        """
+        Args:
+            adj_per_deg: float, energy adjustment to apply per degree K, in eV/atom
+            temp: float, temperature in Kelvin
+            n_atoms: float or int, number of atoms
+            name: str, human-readable name of the energy adjustment.
+                (Default: "")
+            cls: str, the name of the Compatibility class used to generate the
+                energy adjustment. (Default: None)
+            description: str, human-readable explanation of the energy adjustment.
+        """
+        self._value = adj_per_deg
+        self.temp = temp
+        self.n_atoms = n_atoms
+        self.name = name
+        self.cls = cls
+        self.description = description + " ({:.4f} eV/K/atom x {} K x {} atoms)".format(self._value,
+                                                                                        self.temp,
+                                                                                        self.n_atoms,
+                                                                                        )
+
+    @property
+    def value(self):
+        """
+        Return the value of the energy correction in eV.
+        """
+        return self._value * self.temp * self.n_atoms
```

Comment:
@shyamd any thoughts on these `EnergyAdjustment` classes? I'm not real happy with this implementation yet. I don't see how to make a clear distinction between a `ConstantCorrection` and a `CompositionCorrection`, for example, because ANY kind of energy correction we apply will be scaled based on the number of atoms. (For example, if we normalize an entry with composition Fe4O6 to Fe2O3, all corrections, no matter what kind, should get cut in half). So in that sense, everything is composition dependent.

Reply:
In the current Entry, the corrections are absolute. The idea with these EnergyAdjustments is to add metadata on where and how they were computed. So in that sense, an EnergyAdjustment is tightly linked to the ComputedEntry it is contained in. It only has to represent that specific full composition, not any scaling. If you want to scale the composition the EnergyAdjustment has to be recomputed for the new composition as is the case right now.

Reply:
OK, that makes sense. So we need to make sure that `normalize` can scale the values of every type of `EnergyAdjustment` (see comment below).